sycl/source/device_selector.cpp

```diff
@@ -92,16 +99,16 @@ int default_selector::operator()(const device &dev) const {
 
   int Score = REJECT_DEVICE_SCORE;
 
-  // Give preference to device of SYCL BE.
-  if (isDeviceOfPreferredSyclBe(dev))
-    Score = 50;
-
   // override always wins
   if (dev.get_info<info::device::device_type>() == detail::get_forced_type())
     Score += 1000;
 
-  if (dev.is_gpu())
+  if (dev.is_gpu()) {
     Score += 500;
+    // Give preference to device of SYCL BE.
+    if (isDeviceOfPreferredSyclBe(dev))
```

Comment:
I think preferred BE should work for all devices, not only for GPU.

Reply:
The current usage of this function is only used to give a bonus point for the device that is matched with SYCL_BE. 
We don't need to give a bonus point for non-GPU device at this moment.

Reply:
@romanovvlad 
Current options for SYCL_BE are useful only for GPU devices.
When we have multiple BE options for other devices, we can remove this condition later.